src/App.tsx

```diff
@@ -1,85 +1,93 @@
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+type State = {
+  status: string;
+};
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component<{}, State> {
+  state = {
+    status: 'Jam',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  addGood = (good: string) => {
+    this.setState({ status: good });
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  removeGood = () => {
+    this.setState({ status: '' });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { status } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {!status ? (<h1 className="title">No goods selected</h1>) : ''}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        {status ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${status} is selected`}
 
-        <tr data-cy="Good">
-          <td>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => (
+                this.setState({ status: '' })
+              )}
+            />
+          </h1>
+        )
+          : ('')}
+
+        <table className="table">
+          <tbody>
+            {goods.map((good) => (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={classNames({
+                  'has-background-success-light': status === good,
+                })}
+              >
+                <td>
+                  {status === good ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => this.removeGood()}
```

Comment:
```suggestion
                      onClick={this.removeGood}
```

Reply:
No need to create extra wrapper, just pass callback if you don't need to pass any arguments to it